File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        new = s.lower()
        
        i = 0
        j = len(new) - 1

        while(i < j):
            while(i < j and not new[i].isalnum()):
                i+=1
            while(i < j and not new[j].isalnum()):
                j-=1
            if(new[i] != new[j]):
                return False
            i+=1
            j-=1
        return True

File: Data Structures & Algorithms/guess-number-higher-or-lower/submission-0.java
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int lo = 1;
        int hi = n;
        int mid = lo + (hi - lo) / 2;

        int guessRes = guess(mid);

        while(lo <= hi){
            if(guessRes == 0) return mid;
            else if(guessRes == -1) hi = mid - 1;
            else lo = mid + 1;
            mid = lo + (hi - lo) / 2;
            guessRes = guess(mid);
        }

        return -1;
    }
}